plugins/inputs/jboss/jboss.go

```diff
@@ -0,0 +1,1222 @@
+	if id != "" {
+		id = id + "_"
+	}
+	for k, i := range item {
+		switch i.(type) {
+		case int:
+			fields[id+k] = i.(float64)
+		case float64:
+			fields[id+k] = i.(float64)
+		case map[string]interface{}:
+			h.flatten(i.(map[string]interface{}), fields, id+k)
+		default:
+		}
+	}
+}
+
+//func (j *JBoss) prepareRequest(optype int, adress map[string]interface{}) (map[string]interface{}, error) {
+func (h *JBoss) prepareRequest(optype int, adress OrderedMap) (map[string]interface{}, error) {
+	bodyContent := make(map[string]interface{})
+
+	// Create bodyContent
+	switch optype {
+	case GetHosts:
+		bodyContent["operation"] = "read-children-names"
+		bodyContent["child-type"] = "host"
+		bodyContent["address"] = []string{}
+		bodyContent["json.pretty"] = 1
+	case GetServers:
+		bodyContent["operation"] = "read-children-names"
+		bodyContent["child-type"] = "server"
+		bodyContent["recursive-depth"] = 0
+		bodyContent["address"] = adress
+		bodyContent["json.pretty"] = 1
+	case GetDBStat:
+		bodyContent["operation"] = "read-resource"
+		bodyContent["include-runtime"] = "true"
+		bodyContent["recursive-depth"] = 2
+		bodyContent["address"] = adress
+		bodyContent["json.pretty"] = 1
+	case GetJVMStat:
+		bodyContent["operation"] = "read-resource"
+		bodyContent["include-runtime"] = "true"
+		bodyContent["recursive"] = "true"
+		bodyContent["address"] = adress
+		bodyContent["json.pretty"] = 1
+	case GetDeployments:
+		bodyContent["operation"] = "read-children-names"
+		bodyContent["child-type"] = "deployment"
+		bodyContent["address"] = adress
+		bodyContent["json.pretty"] = 1
+	case GetDeploymentStat:
+		bodyContent["operation"] = "read-resource"
+		bodyContent["include-runtime"] = "true"
+		bodyContent["recursive-depth"] = 3
+		bodyContent["address"] = adress
+		bodyContent["json.pretty"] = 1
+	case GetWebStat:
+		bodyContent["operation"] = "read-resource"
+		bodyContent["include-runtime"] = "true"
+		bodyContent["recursive-depth"] = 0
+		bodyContent["address"] = adress
+		bodyContent["json.pretty"] = 1
+	case GetJMSQueueStat:
+		bodyContent["operation"] = "read-children-resources"
+		bodyContent["child-type"] = "jms-queue"
+		bodyContent["include-runtime"] = "true"
+		bodyContent["recursive-depth"] = 2
+		bodyContent["address"] = adress
+		bodyContent["json.pretty"] = 1
+	case GetJMSTopicStat:
+		bodyContent["operation"] = "read-children-resources"
+		bodyContent["child-type"] = "jms-topic"
+		bodyContent["include-runtime"] = "true"
+		bodyContent["recursive-depth"] = 2
+		bodyContent["address"] = adress
+		bodyContent["json.pretty"] = 1
+	}
+
+	return bodyContent, nil
+}
+
+func (j *JBoss) doRequest(domainUrl string, bodyContent map[string]interface{}) ([]byte, error) {
+
+	serverUrl, err := url.Parse(domainUrl)
+	if err != nil {
+		return nil, err
+	}
+	requestBody, err := json.Marshal(bodyContent)
+	method := "POST"
+
+	// Debug JSON request
+	log.Printf("D! Req: %s\n", requestBody)
```

Comment:
I feel like the debug logging in this function to too sloppy, we can leave it in if we make sure it is readable and understandable, otherwise remove it.

Reply:
Yes it is , and helped me to understand all the jboss management protocol , iture  will help if somebody has a particular jboss installation and having troubles gathering certain metrics.